File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
static const bool Booster = [](){
    #pragma GCC optimize("OFast")
    std::ios_base::sync_with_stdio(0);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    return true;
}();

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {

        int minLength   = INT_MAX;
        int totalSum    = 0;
        int leftPointer = 0;

        for(int i=0; i<nums.size(); i++){
            totalSum += nums[i];
            while(totalSum >= target){
                minLength = min(minLength, (i - leftPointer) + 1);
                totalSum = totalSum - nums[leftPointer];
                leftPointer++;
            }
        }

        if(minLength == INT_MAX){
            return 0;
        }
        
        return minLength;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:

        leftPointer = 0
        rightPointer = 0

        currentSum = 0
        minLength = float("inf")

        while rightPointer <= len(nums)-1:

            currentSum += nums[rightPointer]

            while currentSum >= target:
                minLength = min(minLength, rightPointer-leftPointer+1)
                currentSum -= nums[leftPointer]
                leftPointer += 1
                
            rightPointer += 1

        if minLength == float("inf"):
            return 0

        return minLength